tests/Test.Xive/Hive/MemorizedHiveTests.cs

```diff
@@ -83,6 +83,16 @@ public void ShiftsScope()
             Assert.Empty(shifted.Catalog().List());
         }
 
+        [Fact]
+        public void RejectsEmptyScope()
+        {
+            var mem = new RamMnemonic();
+            var hive = new MemorizedHive("", mem);
+            Assert.Throws<ArgumentException>(()=>
+                hive.Catalog()
+            );
```

Comment:
💅🏻 you could also test for the exception message to be sure the correct exception is thrown. but in this case it would be 💅🏻💅🏻

Reply:
I use these:
`namespace Xunit;

/// <summary>
/// Allows deeper investigation of Exceptions.
/// </summary>
public static class AssertException
{
    /// <summary>
    /// Asserts that exception type matches and message equals given one.
    /// </summary>
    public static void MessageMatches<Ex>(string message, Action act)
    {
        try{ act(); }
        catch(Exception ex)
        {
            Assert.IsType<Ex>(ex);
            Assert.Equal(ex.Message, message);
        }
    }

    /// <summary>
    /// Asserts that exception type matches and message starts with given text.
    /// </summary>
    public static void MessageStartsWith<Ex>(string message, Action act)
    {
        try { act(); }
        catch (Exception ex)
        {
            Assert.IsType<Ex>(ex);
            Assert.StartsWith(message, ex.Message);
        }
    }

    /// <summary>
    /// Asserts that exception type matches and message contains given text.
    /// </summary>
    public static void MessageContains<Ex>(string message, Action act)
    {
        try { act(); }
        catch (Exception ex)
        {
            Assert.IsType<Ex>(ex);
            Assert.Contains(message, ex.Message);
        }
    }
}`